htlcswitch/switch.go

```diff
@@ -1952,6 +1960,10 @@ func (s *Switch) addLiveLink(link ChannelLink) {
 		s.interfaceIndex[peerPub] = make(map[lnwire.ChannelID]ChannelLink)
 	}
 	s.interfaceIndex[peerPub][link.ChanID()] = link
+
+	// Since the link addition has succeeded, this channel is now officially
+	// active and we can tell the ChannelNotifier about it.
+	s.cfg.NotifyActiveChannelEvent(link.ChannelState())
```

Comment:
A channel only becomes active once this method returns true: https://github.com/lightningnetwork/lnd/blob/master/htlcswitch/link.go#L459

Reply:
fixed if i interpreted this correctly..